userspace/libpman/src/ringbuffer.c

```diff
@@ -201,13 +202,158 @@ int pman_finalize_ringbuf_array_after_loading()
 	return err;
 }
 
-static void *ringbuf__search_one_from_ring(struct ring *r, int ring_cnt)
+static inline void *ringbuf__get_first_ring_event(struct ring *r, int pos)
 {
-	int *len_ptr, len;
+	int *len_ptr = NULL;
+	int len = 0;
+	void *sample = NULL;
+
+	/* If the consumer reaches the producer update its position. */
```

Comment:
The comment sounds like we are updating the consumer position, but the following line is updating the producer position.
Which is correct? :)

Reply:
yeah, probably the comment is not clear I will update it. When we have consumed all available data we update the producer position to process the new collected events :) so here we update the producer position because the consumer has reached its position :)